### ISSUE TITLE
False positive in `LogicConditionNeedOptimization` on instanceof with pattern variable

### ISSUE THREAD

**Issue by nmancus1** [sevntu-checkstyle/sevntu.checkstyle#882]
From checkstyle/checkstyle#11100

Discovered in report at https://nmancus1.github.io/issue-10848/issue-10848_check_diff_reports_2021_12_31/diff_sevntu-check-regression_part_2/index.html

```bash
➜  javac --enable-preview --release 17 Test.java

➜  cat Test.java
public class Test {
    void m1(Object o) {
        for (int i = 0; o instanceof Integer myInt && myInt > 5;) {
        }
    }
}

➜  cat Test.java
public class Test {
    void m1(Object o) {
        for (int i = 0; o instanceof Integer myInt && myInt > 5;) { // should be ok
        }
    }
}

➜  cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="com.github.sevntu.checkstyle.checks.coding.LogicConditionNeedOptimizationCheck"/>
    </module>
</module>

➜  java -classpath sevntu-checks-1.41.0.jar:checkstyle-9.3-SNAPSHOT-all.jar com.puppycrawl.tools.checkstyle.Main -c config.xml Test.java
Starting audit...
[ERROR] Test.java:3:52: Condition with && at line 3 position 51 need optimization. All method calls are advised to move to end of logic expression. [LogicConditionNeedOptimization]
Audit done.
Checkstyle ends with 1 errors.

```

But, if we follow advice of LogicConditionNeedOptimization check:

```
➜  cat Test.java                                                                                                                        
public class Test {
    void m1(Object o) {
        for (int i = 0; myInt > 5 && o instanceof Integer myInt;) { // should be ok
        }
    }
}

➜  javac --enable-preview --release 17 Test.java                                                                                        
Test.java:3: error: cannot find symbol
        for (int i = 0; myInt > 5 && o instanceof Integer myInt;) { // should be ok
                        ^
  symbol:   variable myInt
  location: class Test
1 error


```

So, this is a false positive.
